tests/test_downloadermiddleware_retry.py

```diff
@@ -103,6 +103,41 @@ def _test_retry_exception(self, req, exception):
         req = self.mw.process_exception(req, exception, self.spider)
         self.assertEqual(req, None)
 
+    def test_different_retry(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url', meta={'max_retry_times': 3})
+
+        # SETINGS: meta(max_retry_times) = 3, RETRY_TIMES = 2
+        self._test_retry(req, DNSLookupError('foo'), 3)
+
+        req2 = Request('http://www.scrapytest.org/invalid_url')
+        
```

Comment:
I find the tests with `req2` difficult to follow with inline comments.
They seem to test the case for changing `RETRY_TIMES` setting only.
So the new meta key and the setting combinations are not tested much apart from `meta(max_retry_times) = 3` and  `RETRY_TIMES = 2`

I suggest that you create a different testcase class than `RetryTest(unittest.TestCase)`,
with each individual test method using a different couple of `RETRY_TIMES` and `meta['max_retry_times']`, initializing a new middleware instance, and checking that `meta['max_retry_times']` takes precedence in all cases where it is set (this is what we want I think). Testing the nominal "without `meta['max_retry_times']`" case is also important of course.
What do you think?

Using seperate test methods within a TestCase class helps when identifying regressions: you can spot 1 test that fails more easily (when running tox with `-v`) than checking which line of code the test broke in a longer test method.
This is not consistent in scrapy tests I admit (and I myself have written long test methods that do not follow what I write here).